spark/src/test/scala/org/apache/comet/CometExpressionSuite.scala

```diff
@@ -2734,4 +2734,28 @@ class CometExpressionSuite extends CometTestBase with AdaptiveSparkPlanHelper {
     }
   }
 
+  test("test integral divide overflow for decimal") {
+    // decimal support requires Spark 3.4 or later
+    assume(isSpark34Plus)
+    if (isSpark40Plus) {
+      Seq(true, false)
+    } else
+      {
+        // ansi mode only supported in Spark 4.0+
```

Comment:
ANSI mode is supported by all 3.x versions as well, but I think it would be ok for Comet to only support ANSI mode with Spark 4 since this is the version that enables it by default